File: backEnd/src/main/java/com/quiz/ourClass/domain/challenge/entity/Challenge.java
package com.quiz.ourClass.domain.challenge.entity;

import com.quiz.ourClass.domain.organization.entity.Organization;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import java.time.LocalDateTime;

@Entity
public class Challenge {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    long id;
    @ManyToOne(fetch = FetchType.LAZY)
    Organization organization;
    String title;
    String content;
    boolean progressStatus;
    int reward;
    LocalDateTime startTime;
    LocalDateTime endTime;
    int minCount;
    int headCount;
}

File: android/core/model/src/main/java/com/sixkids/model/Challenge.kt
package com.sixkids.model

import java.time.LocalDateTime

data class Challenge(
    val id: Long = 0,
    val title: String = "",
    val content: String = "",
    val headCount: Int = 0,
    val reward: Int = 0,
    val startTime: LocalDateTime = LocalDateTime.now(),
    val endTime: LocalDateTime = LocalDateTime.now(),
)
